target/linux/brcm63xx/image/bcm63xx.mk

```diff
@@ -501,7 +543,9 @@ TARGET_DEVICES += DSL2640B-B
 
 define Device/DSL2640U
   $(Device/bcm63xx)
-  DEVICE_TITLE := D-Link DSL-2640U/BRU/C
+  DEVICE_VENDOR := D-Link
+  DEVICE_MODEL := DSL-2640U/BRU
+  DEVICE_VARIANT := C1
```

Comment:
This change is not immediately logical to me. For me it looks more like three alternate model names.
Can somebody enlighten me?

Reply:
Yeah, three different devices, or rather the same device branded for three different ISPs IIRC.

Reply:
So technically this should become three names then.

Reply:
I got the C1 from here: https://github.com/openwrt/openwrt/commit/30a579c6744eab2184689b680424be49369d7ad6#diff-442e09c1bf4842da9c15c05633d8e4f2R302

If it is three devices, is it DSL-2640U , DSL-2640BRU and DSL-2640C?

Reply:
> If it is three devices, is it DSL-2640U , DSL-2640BRU and DSL-2640C?

That's how I meant/understood it.

> I got the C1 from here: 30a579c#diff-442e09c1bf4842da9c15c05633d8e4f2R302

Interesting. No clue what we should make out of that. (All of them C1? Copy/paste error? Something else?)

Reply:
@KanjiMonster ping

Reply:
I don't speak any russian, but it seems this is just one device, and DSL-2640U/BRU/C is the Russian market name (http://www.dlink.ru/ru/products/3/1620.html) - while the English name is DSL-2640U (http://www.dlink.ru/en/products/3/1620.html) ("H/W Ver.: C1" probably).

There does seem to be a DSL-2640C (at least google found an emulator for it), but it seems unrelated at a first glance.

Reply:
Might also be possible that there is just one web page for those different variants.

Reply:
The "Order info" tab for the "DSL-2640U/BRU" also describes that model as "DSL-2600U/BRU/C2"

Reply:
https://wikidevi.com/wiki/D-Link_DSL-2640U_rev_C2
http://ftp.dlink.ru/pub/ADSL/DSL-2640U_BRU_C2/Data_sh/DS_DSL-2640U_BRU_C2_RUS_01.pdf

How about this?
```
DEVICE_VENDOR := D-Link
DEVICE_MODEL := DSL-2640U
DEVICE_VARIANT := C1
DEVICE_ALT0_VENDOR := D-Link
DEVICE_ALT0_MODEL := DSL-2640U/BRU/C
```

Reply:
Hmm, I do not know how reliable names on wikidevi are.
However, based on what I find there and what we've seen so far I think mwarning's latest suggestion is indeed the best one at the moment.

Reply:
Based on wikidevi, "DSL-2640U C2" aka "DSL-2640U/BRU/C2" would be a different device then, so we should ignore it unless we have tests about that.

Reply:
ok, applied.